plugin/src/main/java/org/opensearch/ml/autoredeploy/MLModelAutoReDeployer.java

```diff
@@ -263,6 +265,14 @@ private void triggerModelRedeploy(ModelAutoRedeployArrangement modelAutoRedeploy
         String functionName = (String) Optional
             .ofNullable(sourceAsMap.get(MLModel.FUNCTION_NAME_FIELD))
             .orElse(sourceAsMap.get(MLModel.ALGORITHM_FIELD));
+        if (functionName == null) {
+            log
+                .error(
+                    "Model function_name or algorithm is null, model is not in correct status, please check the model, model id is: {}",
```

Comment:
Does model deployment depends on model function name?

Reply:
It doesn't at the beginning, but in this PR: #2976, we filtering out the remote model, that why we check function name here. And any model should have function name even auto redeploy doesn't depend on that: https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelInput.java#L413